src/app/components/admin/page-admin-articles/page-admin-articles.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Article } from 'src/api/backend';
+import { IconService, RoutesService, YearsService } from 'src/app/services';
+import { AdminArticlesService } from 'src/app/services/admin/admin-articles.service';
+import { AdminWavesService } from 'src/app/services/admin/admin-waves.service';
+
+@Component({
+  selector: 'ksi-page-admin-articles',
+  templateUrl: './page-admin-articles.component.html',
+  styleUrls: ['./page-admin-articles.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PageAdminArticlesComponent implements OnInit {
+
+  constructor(
+    public icon: IconService,
+    public routes: RoutesService,
+    public years: YearsService,
+    private cdr: ChangeDetectorRef,
+    private adminArticlesService: AdminArticlesService
+  ) { }
+
+  articles$: Observable<Article[]>;
+
+
+  ngOnInit(): void {
+    this.reloadArticles();
+  }
+
+  reloadArticles() {
+    this.articles$ = this.adminArticlesService.getArticles();
+    this.cdr.markForCheck();
+  }
+
+  notImplemented(): void {
+    alert(`Feature is not implemented yet.`);
+  }
+
+  deleteArticle(article: Article): void {
+    if (confirm(`Are you sure you want to delete article "${article.title}"?`)) {
+      this.adminArticlesService.deleteArticle(article.id).subscribe({
+        next: () => {
+          this.reloadArticles();
+        },
+        error: (err) => {
+          alert(`Error deleting article: ${err?.message || err}`);
```

Comment:
Místo `alert` prosím používej this.modal.yesNo('discussion-thread.post.delete.confirmation') (viz https://github.com/fi-ksi/web-frontend-angular/blob/main/src/app/components/shared/discussion-thread/discussion-thread-posts/discussion-thread-posts.component.ts#L150 ), je to lépe nastylované. Případně zkus vytvořit vlastní modal utilitku jen na zobrazování popupů.